### PR TITLE
Set remotes in description

### PR DESCRIPTION
**Pull request by kiwiroy**
1. Update the `DESCRIPTION` file to refer to Bioconductor `Remotes` so that the install instructions work and minor typo.
2. Added GitHub Actions to test (1) and run `rcmdcheck`
3. Modify examples so `rmcdcheck` passes them.

### CODE REVIEW

File: .github/workflows/R-CMD-check.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  R-CMD-check:
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@v1
+      - name: Install dependencies
+        run: |
+          install.packages(c("remotes", "rcmdcheck"))
+          remotes::install_deps(dependencies = TRUE)
```
Comment (FelixErnst) [inline .github/workflows/R-CMD-check.yaml:22]: works fine with `remotes::install_deps(dependencies = TRUE, repos = BiocManager::repositories())`.

Using this, the changes in the DESCRIPTION file shouldn't be necessary

File: DESCRIPTION [hunk 2]
```diff
@@ -42,5 +42,9 @@ Collate:
   'RevigoR-plot.R'
   'RevigoR-utils.R'
   'RevigoRData-class.R'
+Remotes:
```
Comment (FelixErnst) [inline DESCRIPTION:45]: See comment above

File: R/RevigoR-RDAVIDWebService.R [hunk 3]
```diff
@@ -17,10 +17,12 @@ NULL
 #' @export
 #'
 #' @examples
+#' \dontrun{
```
Comment (FelixErnst) [inline R/RevigoR-RDAVIDWebService.R:20]: They don't work anymore? Why? It is not a good idea to include dontrun statements here

File: R/RevigoR-plot.R [hunk 4]
```diff
@@ -29,10 +29,12 @@ NULL
 #' @importFrom ggthemes solarized_pal
 #'
 #' @examples
+#' \dontrun{
```
Comment (FelixErnst) [inline R/RevigoR-plot.R:32]: See comment above

File: .github/workflows/R-CMD-check.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+name: R-CMD-check
+
+jobs:
+  R-CMD-check:
+    runs-on: macOS-latest
```
Comment (FelixErnst) [inline .github/workflows/R-CMD-check.yaml:15]: Please include a linux build or switch to linux entirely. Thanks

**Review by FelixErnst — CHANGES_REQUESTED** [review]
Hi Roy @kiwiroy ,

thanks for the PR. Since I don't have the permission of the website hoster to put this in BioC, I am not actively developing here. 

However, If you fix the issues mentioned in the comments, I am happy to merge the PR:

Felix

**Review by FelixErnst — APPROVED** [review]

File: .github/workflows/R-CMD-check.yaml (hunk 1 above)
Comment (kiwiroy) [reply .github/workflows/R-CMD-check.yaml]: Done, `README.md` updated to reflect it.

File: R/RevigoR-RDAVIDWebService.R (hunk 3 above)
Comment (kiwiroy) [reply R/RevigoR-RDAVIDWebService.R]: Yeah, they didn't, but now they do, looks like it was a combination of X11 error and website error. Added `http_error` test.

File: R/RevigoR-plot.R (hunk 4 above)
Comment (kiwiroy) [reply R/RevigoR-plot.R]: As above

File: .github/workflows/R-CMD-check.yaml (hunk 5 above)
Comment (kiwiroy) [reply .github/workflows/R-CMD-check.yaml:15]: Done.

File: DESCRIPTION (hunk 2 above)
Comment (kiwiroy) [reply DESCRIPTION]: Done.

**Comment by kiwiroy** [comment]
@FelixErnst I resolved all the suggestions you've made. Revigo web service was not stable yesterday confounding my installation woes, hopefully I've made it a little more obvious when the service isn't responding.  Thanks for the codes.

**Pull request merged**